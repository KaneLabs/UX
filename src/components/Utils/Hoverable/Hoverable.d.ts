export default Hoverable;
declare class Hoverable extends React.Component<any, any, any> {
    static displayName: string;
    static propTypes: {
        children: PropTypes.Requireable<string | number | boolean | {} | PropTypes.ReactElementLike | PropTypes.ReactNodeArray>;
        onHoverIn: PropTypes.Requireable<(...args: any[]) => any>;
        onHoverOut: PropTypes.Requireable<(...args: any[]) => any>;
        testID: PropTypes.Requireable<string>;
    };
    constructor(props: any);
    constructor(props: any, context: any);
    _handleMouseEnter: () => void;
    _handleMouseLeave: () => void;
    _toggle: () => void;
}
import React from "react";
import PropTypes from "prop-types";
