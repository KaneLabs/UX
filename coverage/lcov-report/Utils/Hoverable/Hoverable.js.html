
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Utils/Hoverable/Hoverable.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">Utils/Hoverable</a> Hoverable.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">68.92% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>51/74</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.33% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>8/24</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>6/12</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">5x</span></td><td class="text"><pre class="prettyprint lang-js">var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));var _createClass2=_interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _inherits2=_interopRequireDefault(require("@babel/runtime/helpers/inherits"));var _possibleConstructorReturn2=_interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));var _getPrototypeOf2=_interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));var _react=_interopRequireWildcard(require("react"));var _reactNative=require("react-native");var _propTypes=_interopRequireDefault(require("prop-types"));var _jsxFileName="/Users/kane/Lab/Eros/src/components/Utils/Hoverable/Hoverable.js";function _createSuper(Derived){var hasNativeReflectConstruct=_isNativeReflectConstruct();return function _createSuperInternal(){var Super=(0,_getPrototypeOf2.default)(Derived),result;<span class="missing-if-branch" title="else path not taken" >E</span>if(hasNativeReflectConstruct){var NewTarget=(0,_getPrototypeOf2.default)(this).constructor;result=Reflect.construct(Super,arguments,NewTarget);}else{<span class="cstat-no" title="statement not covered" >result=Super.apply(this,arguments);}</span>return(0,_possibleConstructorReturn2.default)(this,result);};}function _isNativeReflectConstruct(){<span class="missing-if-branch" title="if path not taken" >I</span>if(typeof Reflect==="undefined"||!Reflect.construct)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="if path not taken" >I</span>i</span>f(Reflect.construct.sham)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="else path not taken" >E</span>i</span>f(typeof Proxy==="function")return true;<span class="cstat-no" title="statement not covered" >try{<span class="cstat-no" title="statement not covered" >Boolean.prototype.valueOf.call(Reflect.construct(Boolean,[],<span class="fstat-no" title="function not covered" >fu</span>nction(){}));<span class="cstat-no" title="statement not covered" >r</span>eturn true;}</span>catch(e){<span class="cstat-no" title="statement not covered" >return false;}</span>}</span>var Hoverable=function(_Component){(0,_inherits2.default)(Hoverable,_Component);var _super=_createSuper(Hoverable);function Hoverable(){var _this;(0,_classCallCheck2.default)(this,Hoverable);for(var _len=arguments.length,args=new Array(_len),_key=0;_key&lt;_len;_key++){args[_key]=arguments[_key];}_this=_super.call.apply(_super,[this].concat(args));_this.state={isHovered:false};_this._handleMouseEnter=<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >if(!_this.state.isHovered){var onHoverIn=<span class="cstat-no" title="statement not covered" >_this.props.onHoverIn;<span class="cstat-no" title="statement not covered" ></span>if(onHoverIn){<span class="cstat-no" title="statement not covered" >onHoverIn();}<span class="cstat-no" title="statement not covered" ></span>_</span>this.setState(<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >return{isHovered:true};}</span>);}</span>}</span>;_this._handleMouseLeave=<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >if(_this.state.isHovered){var onHoverOut=<span class="cstat-no" title="statement not covered" >_this.props.onHoverOut;<span class="cstat-no" title="statement not covered" ></span>if(onHoverOut){<span class="cstat-no" title="statement not covered" >onHoverOut();}<span class="cstat-no" title="statement not covered" ></span>_</span>this.setState(<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >return{isHovered:false};}</span>);}</span>}</span>;_this._toggle=<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >if(_this.state.isHovered){<span class="cstat-no" title="statement not covered" >_this._handleMouseLeave();}</span>else{<span class="cstat-no" title="statement not covered" >_this._handleMouseEnter();}</span>}</span>;return _this;}(0,_createClass2.default)(Hoverable,[{key:"render",value:function render(){var _this$props=this.props,children=_this$props.children,testID=_this$props.testID;var child=typeof children==='function'?<span class="branch-0 cbranch-no" title="branch not covered" >children(this.state.isHovered):</span>children;<span class="missing-if-branch" title="if path not taken" >I</span>if(_reactNative.Platform.OS==='web'){<span class="cstat-no" title="statement not covered" >return _react.default.cloneElement(_react.default.Children.only(child),{onMouseEnter:this._handleMouseEnter,onMouseLeave:this._handleMouseLeave});}</span>return _react.default.createElement(_reactNative.TouchableWithoutFeedback,{onPress:this._toggle,testID:testID,__self:this,__source:{fileName:_jsxFileName,lineNumber:61,columnNumber:7}},_react.default.createElement(_reactNative.View,{__self:this,__source:{fileName:_jsxFileName,lineNumber:62,columnNumber:9}},_react.default.cloneElement(_react.default.Children.only(child),{onMouseEnter:this._handleMouseEnter,onMouseLeave:this._handleMouseLeave})));}}]);return Hoverable;}(_react.Component);Hoverable.displayName='Hoverable';Hoverable.propTypes={children:_propTypes.default.oneOfType([_propTypes.default.node,_propTypes.default.func]),onHoverIn:_propTypes.default.func,onHoverOut:_propTypes.default.func,testID:_propTypes.default.string};var _default=Hoverable;exports.default=_default;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Mar 31 2021 17:15:17 GMT-0600 (Mountain Daylight Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    