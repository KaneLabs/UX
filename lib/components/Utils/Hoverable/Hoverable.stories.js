var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _react=_interopRequireDefault(require("react"));var _reactNative=require("react-native");var _storybookState=require("@sambego/storybook-state");var _=require("../..");var _Header=_interopRequireDefault(require("../../storybook/components/Header"));var _Container=_interopRequireDefault(require("../../storybook/components/Container"));var _storiesOf=require("../../../../storybook/helpers/storiesOf");var _this=this,_jsxFileName="/Users/kane/Lab/Eros/src/components/Utils/Hoverable/Hoverable.stories.js";var store=new _storybookState.Store({color:'#00BCD4',left:new _reactNative.Animated.Value(0)});var _default=(0,_storiesOf.storiesOf)('Utils|Hoverable',module).addParameters({jest:['Hoverable']}).add('Simple',function(){return _react.default.createElement(_Container.default,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:19,columnNumber:5}},_react.default.createElement(_Header.default,{title:"Hoverable",subtitle:"Cross Platform Hover Component",__self:_this,__source:{fileName:_jsxFileName,lineNumber:20,columnNumber:7}}),_react.default.createElement(_storybookState.State,{store:store,style:{flex:1},__self:_this,__source:{fileName:_jsxFileName,lineNumber:21,columnNumber:7}},function(state){return _react.default.createElement(_reactNative.View,{style:{flexDirection:'row'},__self:_this,__source:{fileName:_jsxFileName,lineNumber:23,columnNumber:11}},_react.default.createElement(_.Hoverable,{onHoverIn:function onHoverIn(){return store.set({color:'#009688'});},onHoverOut:function onHoverOut(){return store.set({color:'#00BCD4'});},__self:_this,__source:{fileName:_jsxFileName,lineNumber:24,columnNumber:13}},_react.default.createElement(_reactNative.View,{style:{width:200,height:125,backgroundColor:state.color},__self:_this,__source:{fileName:_jsxFileName,lineNumber:28,columnNumber:15}})));}));}).add('Animation',function(){return _react.default.createElement(_Container.default,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:42,columnNumber:5}},_react.default.createElement(_Header.default,{title:"Hoverable",subtitle:"Cross Platform Hover Component",__self:_this,__source:{fileName:_jsxFileName,lineNumber:43,columnNumber:7}}),_react.default.createElement(_storybookState.State,{store:store,style:{flex:1},__self:_this,__source:{fileName:_jsxFileName,lineNumber:44,columnNumber:7}},function(state){return _react.default.createElement(_reactNative.View,{style:{flexDirection:'row'},__self:_this,__source:{fileName:_jsxFileName,lineNumber:46,columnNumber:11}},_react.default.createElement(_.Hoverable,{onHoverIn:function onHoverIn(){_reactNative.Animated.timing(state.left,{toValue:1,duration:500}).start();},onHoverOut:function onHoverOut(){_reactNative.Animated.timing(state.left,{toValue:0,duration:500}).start();},__self:_this,__source:{fileName:_jsxFileName,lineNumber:47,columnNumber:13}},_react.default.createElement(_reactNative.Animated.View,{useNativeDriver:false,style:{width:200,height:125,backgroundColor:state.color,transform:[{rotate:state.left.interpolate({inputRange:[0,1],outputRange:['0deg','360deg']})}]},__self:_this,__source:{fileName:_jsxFileName,lineNumber:61,columnNumber:15}})));}));});exports.default=_default;