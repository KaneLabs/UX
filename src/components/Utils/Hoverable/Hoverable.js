import React, { Component } from 'react';
import {
  Platform,
  Dimensions,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

class Hoverable extends Component {
  static displayName = 'Hoverable';

  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    onHoverIn: PropTypes.func,
    onHoverOut: PropTypes.func,
    testID: PropTypes.string,
  };

  state = {
    isHovered: false,
  };

  _handleMouseEnter = () => {
    if (!this.state.isHovered) {
      const { onHoverIn } = this.props;
      if (onHoverIn) {
        onHoverIn();
      }
      this.setState(() => ({ isHovered: true }));
    }
  };

  _handleMouseLeave = () => {
    if (this.state.isHovered) {
      const { onHoverOut } = this.props;
      if (onHoverOut) {
        onHoverOut();
      }
      this.setState(() => ({ isHovered: false }));
    }
  };

  _toggle = () => {
    if (this.state.isHovered) {
      this._handleMouseLeave();
    } else {
      this._handleMouseEnter();
    }
  };

  render() {
    const { children, testID } = this.props;

    const child =
      typeof children === 'function'
        ? children(this.state.isHovered)
        : children;

    if (Platform.OS === 'web') {
      return React.cloneElement(React.Children.only(child), {
        onMouseEnter: this._handleMouseEnter,
        onMouseLeave: this._handleMouseLeave,
      });
    }
    return (
      <TouchableWithoutFeedback onPress={this._toggle} testID={testID}>
        <View>
          {React.cloneElement(React.Children.only(child), {
            onMouseEnter: this._handleMouseEnter,
            onMouseLeave: this._handleMouseLeave,
          })}
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

export default Hoverable;
