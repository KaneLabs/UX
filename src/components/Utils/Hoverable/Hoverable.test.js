import React from 'react';
import { View } from 'react-native';
import ShallowRenderer from 'react-test-renderer/shallow';
import Hoverable from './Hoverable';

const renderer = new ShallowRenderer();

test('Switch Renders', () => {
  const tree = renderer.render(
    <Hoverable>
      <View />
    </Hoverable>,
  );
  expect(tree).toMatchSnapshot();
});
