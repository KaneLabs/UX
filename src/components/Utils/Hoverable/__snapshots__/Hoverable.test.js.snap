// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Renders 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View>
    <View
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </View>
</TouchableWithoutFeedback>
`;
